### ISSUE TITLE
Security Group Watcher not properly parsing security groups

### ISSUE THREAD

**Issue by BobPeterson1881** [Netflix/security_monkey#621]
The latest security group watcher script (security_group.py) is not parsing all the access grants correctly.

The script is only returning the first 'IpRanges' and/or 'UserIdGroupPairs' value from the 'IpPermission' and 'IpPermissionEgress' dictionary items.  If there are multiple list items in  'IpRanges' or 'UserIdGroupPairs', those are not being included in the list of rules.  Each item in these lists should be returned as a separate "rule" entry.

This problem was introduced with the switch to boto3 for the watcher scripts.

**Comment by BobPeterson1881** [comment]
I updated the security group watcher script to parse out the IpRanges and UserIdGroups into separate rules.  Works, but there may be a better way!

Changes:
```
    def _build_rule(self, rule, rule_type):
        rule_list=[]
        #base rule information
        rule_config = {
            "ip_protocol": rule.get('IpProtocol'),
            "rule_type": rule_type,
            "from_port": rule.get('FromPort'),
            "to_port": rule.get('ToPort'),
            "cidr_ip": None,
            "owner_id": None,
            "group_id": None,
            "name": None
        }
    
        for ips in rule.get('IpRanges'):
            #make a copy of the base rule info.
            new_rule=rule_config.copy()
            new_rule['cidr_ip'] = ips.get('CidrIp')
            rule_list.append(new_rule)
            
        for user_id_group_pairs in rule.get('UserIdGroupPairs'):
            #make a copy of the base rule info.
            new_rule=rule_config.copy()
            new_rule['owner_id'] = user_id_group_pairs.get('UserId')
            new_rule['group_id'] = user_id_group_pairs.get('GroupId')
            new_rule['name'] = user_id_group_pairs.get('GroupName')
            rule_list.append(new_rule)                
            
        return rule_list
```


in the slurp function I also changed the calls to _build_rule.


```
                    for rule in sg['IpPermissions']:
                        item_config['rules'] += self._build_rule(rule,"ingress")

                    for rule in sg['IpPermissionsEgress']:
                        item_config['rules'] += self._build_rule(rule,"egress")
```

**Comment by monkeysecurity** [comment]
Could you PR that over?

**Comment by BobPeterson1881** [comment]
Done.  New to GitHub, so learning new things!

**Comment by monkeysecurity** [comment]
Fixed in #629
